packages/edit-post/src/components/manage-blocks-modal/manager.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * External dependencies
+ */
+import { filter, includes, map, find, some, deburr } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { withSelect, withDispatch } from '@wordpress/data';
+import { compose, withInstanceId, withState } from '@wordpress/compose';
+import { ToggleControl, PanelBody } from '@wordpress/components';
+import { __experimentalBlockTypesList as BlockTypesList } from '@wordpress/editor';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Filters an item list given a search term.
+ *
+ * @param {Object[]} items      Set of inserter items to search within.
+ * @param {Object[]} categories Set of categories to search within.
+ * @param {string}   searchTerm Search term.
+ *
+ * @return {Object[]} Filtered item list.
+ */
+export const searchItems = ( items, categories, searchTerm ) => {
+	const normalizedSearchTerm = normalizeTerm( searchTerm );
+	const matchSearch = ( string ) => normalizeTerm( string ).includes( normalizedSearchTerm );
+
+	return items.filter( ( item ) => {
+		const itemCategory = find( categories, { slug: item.category } );
+
+		return (
+			matchSearch( item.title ) ||
+			some( item.keywords, matchSearch ) ||
+			( itemCategory && matchSearch( itemCategory.title ) )
+		);
+	} );
+};
+
+/**
+ * Converts the search term into a normalized term.
+ *
+ * @param {string} term The search term to normalize.
+ *
+ * @return {string} The normalized search term.
+ */
+export const normalizeTerm = ( term ) => {
```

Comment:
Should `normalizeTerm` and `searchItems` be extracted to a `block` package (or a different one) to avoid code duplication and ensure they work the same moving forward? 

Isn't it also somehow replicated in the block autocomplete component?

Reply:
> Should `normalizeTerm` and `searchItems` be extracted to a `block` package (or a different one) to avoid code duplication and ensure they work the same moving forward?

Yeah, I wasn't quite sure how I wanted to abstract this yet. To your other comment, it's something I could see becoming part of `BlockTypesList` as part of a common search behavior between the inserter and block manager. I'm not familiar yet if it's the same for autocomplete, but I suppose it makes sense that it would be. And if so, maybe you're right that `blocks` might be a good place for it. I could even imagine it being a selector in the blocks package (`getBlocksBySearchTerm` or otherwise).